unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -5119,47 +5111,41 @@ private IEnumerator checkDropHandObjectActionFast(SimObjPhysics currentHandSimOb
             actionFinished(true);
         }
 
-        // Helper method that parses objectId and (x and y) parameters to return the
-        // sim object that they target.
-        private SimObjPhysics getTargetObject(
-                string objectId,
-                bool forceAction = false
-        ) {
+        // Helper method that parses objectId parameter to return the sim object that it target.
+        // The action is halted if the objectId does not appear in the scene.
+        private SimObjPhysics getTargetObject(string objectId, bool forceAction = false) {
             // an objectId was given, so find that target in the scene if it exists
             if (!physicsSceneManager.ObjectIdToSimObjPhysics.ContainsKey(objectId)) {
-                errorMessage = "Object ID appears to be invalid.";
-                return null;
+                throw new ArgumentException($"objectId: {objectId} is not the objectId on any object in the scene!");
```

Comment:
Using `ArgumentException` to mean anything that is non-null and not bound to a particular numberic range (e.g., x/y must be in [0:1]).